sdk/core/System.ClientModel/src/ModelReaderWriter/ModelJsonConverter.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace System.ClientModel.Primitives;
+
+/// <summary>
+/// A generic converter which allows <see cref="JsonSerializer"/> to be able to write and read any models that implement <see cref="IJsonModel{T}"/>.
+/// </summary>
+#if !NET5_0 // RequiresUnreferencedCode in net5.0 doesn't have AttributeTargets.Class as a target, but it was added in net6.0
+[RequiresUnreferencedCode("The constructors of the type being deserialized are dynamically accessed and may be trimmed.")]
+#endif
+#pragma warning disable AZC0014 // Avoid using banned types in public API
+public class ModelJsonConverter : JsonConverter<IJsonModel<object>>
+#pragma warning restore AZC0014 // Avoid using banned types in public API
+{
+    /// <summary>
+    /// Gets the <see cref="ModelReaderWriterOptions"/> used to read and write models.
+    /// </summary>
+    public ModelReaderWriterOptions Options { get; }
+
+    /// <summary>
+    /// Initializes a new instance of <see cref="ModelJsonConverter"/> with a default options of <see cref="ModelReaderWriterOptions.Json"/>.
+    /// </summary>
+    public ModelJsonConverter()
+        : this(ModelReaderWriterOptions.Json) { }
+
+    /// <summary>
+    /// Initializes a new instance of <see cref="ModelJsonConverter"/>.
+    /// </summary>
+    /// <param name="options">The <see cref="ModelReaderWriterOptions"/> to use.</param>
+    public ModelJsonConverter(ModelReaderWriterOptions options)
+    {
+        Options = options;
+    }
+
+    /// <inheritdoc/>
+    public override bool CanConvert(Type typeToConvert)
+    {
+        return !Attribute.IsDefined(typeToConvert, typeof(JsonConverterAttribute));
+    }
+
+    /// <inheritdoc/>
+#pragma warning disable AZC0014 // Avoid using banned types in public API
```

Comment:
Would explicitly implementing these bypass the need to suppress these warnings?

Reply:
Not sure I follow this is override a virtual method not implementing an interface?